beacon_node/network/src/sync/network_context.rs

```diff
@@ -405,4 +421,93 @@ impl<T: BeaconChainTypes> SyncNetworkContext<T> {
         self.range_blocks_and_blobs_requests
             .insert(id, (sender_id, info));
     }
+
+    // Request handlers
+
+    pub fn on_single_block_response(
+        &mut self,
+        request_id: SingleLookupReqId,
+        block: RpcEvent<Arc<SignedBeaconBlock<T::EthSpec>>>,
+    ) -> RpcProcessingResult<Arc<SignedBeaconBlock<T::EthSpec>>> {
+        let Entry::Occupied(mut request) = self.blocks_by_root_requests.entry(request_id) else {
+            return None;
+        };
+
+        Some(match block {
+            RpcEvent::Response(block, seen_timestamp) => {
+                match request.get_mut().add_response(block) {
+                    Ok(block) => Ok((block, seen_timestamp)),
+                    Err(e) => {
+                        // The request must be dropped after receiving an error.
+                        // TODO: We could NOT drop the request here, and penalize the peer again if
+                        // sends multiple penalizable chunks after the first invalid.
+                        request.remove();
+                        Err(e)
+                    }
+                }
+            }
+            RpcEvent::StreamTermination => match request.remove().terminate() {
+                Ok(_) => return None,
+                Err(e) => Err(e),
+            },
+            RpcEvent::RPCError(e) => {
+                request.remove();
+                Err(e)
+            }
+        })
+    }
+
+    pub fn on_single_blob_response(
+        &mut self,
+        request_id: SingleLookupReqId,
+        blob: RpcEvent<Arc<BlobSidecar<T::EthSpec>>>,
+    ) -> RpcProcessingResult<FixedBlobSidecarList<T::EthSpec>> {
+        let Entry::Occupied(mut request) = self.blobs_by_root_requests.entry(request_id) else {
+            return None;
+        };
+
+        Some(match blob {
+            RpcEvent::Response(blob, _) => match request.get_mut().add_response(blob) {
+                // TODO: Should deal only with Vec<Arc<BlobSidecar>>
```

Comment:
Should this be moved to an issue as well?

Reply:
removed TODO here: https://github.com/dapplion/lighthouse/pull/24/commits/c8ac5a0492a14909eb6521ddaeaef388ea5e8cd9

issue here: https://github.com/sigp/lighthouse/issues/5604